tests/test_weapon_cadence.c

```
@@ -0,0 +1,249 @@
+/* ------------------------------------------------------------------ *
+ * Task B: round-robin subsystem coverage (#194 anti-drift)            *
+ * ------------------------------------------------------------------ */
+
+/*
+ * Drive bc_ship_build_health_update repeatedly, feeding next_idx back as
+ * start_idx (exactly as the 10 Hz broadcaster does). Assert that every
+ * ser_list entry index is visited within the bounded cycle, and that the
+ * cursor returns to its starting point — i.e. the round-robin IS a periodic
+ * full resend of all subsystem state. No separate force-resend timer needed.
+ */
+TEST(round_robin_covers_all_subsystems_within_cycle)
```

Clang (14.0.6)

[warning] function 'test_round_robin_covers_all_subsystems_within_cycle' has cognitive complexity of 43 (threshold 25)

(readability-function-cognitive-complexity)